src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -93,12 +93,22 @@ public User getUser(UUID id) {
     return null;
   }
 
-  /** Add a new user to the current set of users known to the application. */
+  /**
+   * Add a new user to the current set of users known to the application. This should only be called
+   * to add a new user, not to update an existing user.
+   */
   public void addUser(User user) {
     users.add(user);
     persistentStorageAgent.writeThrough(user);
   }
 
+  /**
+   * Update an existing User.
+   */
+  public void updateUser(User user) {
+    persistentStorageAgent.writeThrough(user);
```

Comment:
Shouldn't we also update our own cached object? (Or in alternative re-read the whole thing?)

Reply:
Currently the only way to get a `User` instance is through the live in-memory `List` in `UserStore`. It's a live reference, so updating the reference will update the data in memory.

An example workflow would be to call the `getUser(String username)` function to fetch a `User` instance, to call something like `user.setAbout(String about)` and then pass that `User` into this `updateUser()` function.

Nothing actually calls this function yet, but I wanted to include it to prevent students from calling `addUser()` to modify data.

Reply:
my bad, I was reviewing with a C++ eye. Forgot that in Java everything is a reference :D